tonic/src/request.rs

```diff
@@ -7,6 +8,30 @@ pub struct Request<T> {
     message: T,
 }
 
+/// Conversion into a streaming `Request`.
+///
+/// Types implementing this trait may be used as an argument to
+/// generated client-streaming RPC methods.
+pub trait IntoStreamingRequest: sealed::Sealed {
+    /// The RPC request stream type
+    type Stream: Stream<Item = Self::Message> + Send + 'static;
+
+    /// The RPC request type
+    type Message;
+
+    /// Wrap the stream of messages in a `tonic::Request`
+    fn into_streaming_request(self) -> Request<Self::Stream>;
+}
+
+/// Conversion into a unary `Request`.
```

Comment:
Can we add a bit more docs here to explain how this works? I am thinking maybe an example that shows constructing a request and passing it and one with the message explaining the trait?

Reply:
Could use https://doc.rust-lang.org/stable/std/convert/trait.Into.html as inspiration

Reply:
Done. Docs are slightly better but I'm not sure they are clear enough.